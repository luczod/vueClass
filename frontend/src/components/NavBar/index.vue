<script  lang="js">
import { RouterLink } from 'vue-router';
import { mapState } from 'vuex';

export default {
  name: "NavBar",
  data() {
    return {

    }
  },
  methods: {
    logout(e) {
      e.preventDefault();

      // Emit event to logout user
      this.$store.commit("logout");
      // redirect
      this.$router.push("/");
    }
  },
  computed: {
    ...mapState([
      'authenticated',
    ])
  }
}
</script>

<template>
  <nav id="nav">
    <router-link to="/profile" id="logo-container">
      <img src="/img/partytimelogo.png" alt="Party Time" />
    </router-link>
    <h2 id="nav-title">Party Time</h2>
    <div id="nav-links">
      <!-- <router-link :to="{ name: 'home' }">Home</router-link> -->
      <router-link to="/">Festas</router-link>
      <router-link to="/login" v-show="!authenticated">Entrar</router-link>
      <router-link to="/register" v-show="!authenticated">Cadastrar</router-link>
      <router-link to="/dashboard" v-show="authenticated">Dashboard</router-link>
      <router-link to="/profile" v-show="authenticated">Configurações</router-link>
      <router-link to="/about">About</router-link>
      <router-link to="/register">Cadastrar</router-link>
      <button @click="logout($event)" v-show="authenticated">Logout</button>
    </div>
  </nav>
</template>

<style scoped>
#nav {
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: sticky;
  top: 0;
  padding: 1.2rem;
  background-color: rgb(226 232 240);
  border-bottom: 1px solid #a3a3a3;
}

#nav img {
  width: 2.25rem;
}

#nav-title {
  font-size: 3rem;
  font-weight: 300;
}

#logo-container,
#nav-links {
  width: 25rem;
}

#nav-links {
  display: flex;
  justify-content: flex-end;
}

#nav-links a {
  display: block;
  position: relative;
  padding: 1.2rem;
}

#nav-links a::after {
  content: '';
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #c1b696;
  transition: all 300ms ease-in-out;
}

#nav-links a:hover::after {
  left: 25%;
  width: 50%;
}

/* button {
  border-color: #fff;
  border: none;
  font-size: 1rem;
  cursor: pointer;
} */

button {
  width: fit-content;
  text-transform: uppercase;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  cursor: pointer;
  border-radius: 0.8rem;
  color: black;
  background-color: transparent;
  border: 0;
  transition: 200ms;
}

button:hover {
  color: #c1b696;
}
</style>


