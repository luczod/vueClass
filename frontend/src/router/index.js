import { createRouter, createWebHistory } from 'vue-router';
import store from '../store';

const routes = [
  {
    path: '/',
    name: 'Home',
    component: () => import('../views/HomePage.vue'),
    meta: {
      requiresAuth: false,
    },
  },
  {
    path: '/about',
    name: 'About',
    component: () => import('../views/About.vue'),
    meta: {
      requiresAuth: false,
    },
  },

  {
    path: '/register',
    name: 'Register',
    component: () => import('../views/Register.vue'),
    meta: {
      requiresAuth: false,
    },
  },
  {
    path: '/login',
    name: 'Login',
    component: () => import('../views/Login.vue'),
    meta: {
      requiresAuth: false,
    },
  },
  {
    path: '/profile',
    name: 'Profile',
    component: () => import('../views/Profile.vue'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/dashboard',
    name: 'Dashboard',
    component: () => import('../views/Dashboard.vue'),
    meta: {
      requiresAuth: true,
    },
  },

  {
    path: '/newparty',
    name: 'NewParty',
    component: () => import('../views/NewParty.vue'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/party/:id',
    name: 'Party',
    component: () => import('../views/Party.vue'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/editparty/:id',
    name: 'Edit Party',
    component: () => import('../views/EditParty.vue'),
    meta: {
      requiresAuth: true,
    },
  },
];

const router = createRouter({
  routes,
  history: createWebHistory(),
});

router.beforeEach((to, from, next) => {
  if (to.matched.some((record) => record.meta.requiresAuth)) {
    if (store.getters.authenticated === false) {
      next({
        path: '/login',
        params: { nextUrl: to.fullPath },
      });
    } else {
      next();
    }
  } else {
    next();
  }
});

export default router;
