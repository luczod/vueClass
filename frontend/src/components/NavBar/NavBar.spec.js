import { describe, expect, it, vi } from 'vitest';
import { mount } from '@vue/test-utils';
import router from '../../router';
import NavBar from './index.vue';

describe('<NavBar />', () => {
  const wrapper = mount(NavBar, {
    global: {
      plugins: [router],
    },
  });

  it('should be defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('link to view homePage', async () => {
    const push = vi.spyOn(router, 'push');
    await wrapper.find('a[href="/"]').trigger('click');

    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith('/');
  });

  it('link to view About', async () => {
    const push = vi.spyOn(router, 'push');
    await wrapper.find('a[href="/about"]').trigger('click');

    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith('/about');
  });

  it('link to view Register', async () => {
    const push = vi.spyOn(router, 'push');
    await wrapper.find('a[href="/register"]').trigger('click');

    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith('/register');
  });

  it('should match rendered', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
