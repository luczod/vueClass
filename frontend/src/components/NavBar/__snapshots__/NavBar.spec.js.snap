// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NavBar /> > should match rendered 1`] = `
<nav
  data-v-61681749=""
  id="nav"
>
  <a
    class=""
    data-v-61681749=""
    href="/profile"
    id="logo-container"
  >
    <img
      alt="Party Time"
      data-v-61681749=""
      src="/img/partytimelogo.png"
    />
  </a>
  <h2
    data-v-61681749=""
    id="nav-title"
  >
    Party Time
  </h2>
  <div
    data-v-61681749=""
    id="nav-links"
  >
    <!-- &lt;router-link :to="{ name: 'home' }"&gt;Home&lt;/router-link&gt; -->
    <a
      class=""
      data-v-61681749=""
      href="/"
    >
      Home
    </a>
    <a
      class=""
      data-v-61681749=""
      href="/about"
    >
      About
    </a>
    <a
      class=""
      data-v-61681749=""
      href="/register"
    >
      Cadastrar
    </a>
  </div>
</nav>
`;
